qcelemental/tests/test_molparse_from_string.py

```diff
@@ -0,0 +1,1136 @@
+import sys
+import copy
+#import pprint
+
+import pytest
+import numpy as np
+
+from utils_compare import *
+#from addons import *
+
+import qcelemental
+
+#import qcdb
+
+subject1 = """O 0 0   0
+no_com
+
+H 1 ,, 0 \t  0 # stuff-n-nonsense"""
+
+ans1 = {
+    'geom': [0., 0., 0., 1., 0., 0.],
+    'elbl': ['O', 'H'],
+    'fix_com': True,
+    'fragment_separators': [],
+    'fragment_charges': [None],
+    'fragment_multiplicities': [None],
+    'fragment_files': [],
+    'geom_hints': [],
+    'hint_types': [],
+}
+
+fullans1a = {
+    'geom': np.array([0., 0., 0., 1., 0., 0.]),
+    'elea': np.array([16, 1]),
+    'elez': np.array([8, 1]),
+    'elem': np.array(['O', 'H']),
+    'mass': np.array([15.99491462, 1.00782503]),
+    'real': np.array([True, True]),
+    'elbl': np.array(['', '']),
+    'units': 'Angstrom',
+    'fix_com': True,
+    'fix_orientation': False,
+    'fragment_separators': [],
+    'fragment_charges': [0.0],
+    'fragment_multiplicities': [2],
+    'molecular_charge': 0.0,
+    'molecular_multiplicity': 2,
+}
+fullans1c = copy.deepcopy(fullans1a)
+fullans1c.update({
+    'fragment_charges': [1.],
+    'fragment_multiplicities': [1],
+    'molecular_charge': 1.,
+    'molecular_multiplicity': 1
+})
+
+
+def test_psi4_qm_1a():
+    subject = subject1
+
+    final, intermed = qcelemental.molparse.from_string(subject, return_processed=True)
+    assert compare_dicts(ans1, intermed, 4, sys._getframe().f_code.co_name + ': intermediate')
+    assert compare_molrecs(fullans1a, final['qm'], 4, sys._getframe().f_code.co_name + ': full')
+
+
+def test_psi4_qm_1b():
+    subject = '\n' + '\t' + subject1 + '\n\n'
+
+    final, intermed = qcelemental.molparse.from_string(subject, return_processed=True)
+    assert compare_dicts(ans1, intermed, 4, sys._getframe().f_code.co_name + ': intermediate')
+    assert compare_molrecs(fullans1a, final['qm'], 4, sys._getframe().f_code.co_name + ': full')
+
+
+def test_psi4_qm_1c():
+    subject = '1 1\n  -- \n' + subject1
+    ans = copy.deepcopy(ans1)
+    ans.update({'molecular_charge': 1., 'molecular_multiplicity': 1})
+
+    final, intermed = qcelemental.molparse.from_string(subject, return_processed=True)
+    assert compare_dicts(ans, intermed, 4, sys._getframe().f_code.co_name + ': intermediate')
+    assert compare_molrecs(fullans1c, final['qm'], 4, sys._getframe().f_code.co_name + ': full')
+
+
+def test_psi4_qm_1d():
+    subject = subject1 + '\n1 1'
+    ans = copy.deepcopy(ans1)
+    ans.update({'fragment_charges': [1.], 'fragment_multiplicities': [1]})
+
+    final, intermed = qcelemental.molparse.from_string(subject, return_processed=True)
+    assert compare_dicts(ans, intermed, 4, sys._getframe().f_code.co_name + ': intermediate')
+    assert compare_molrecs(fullans1c, final['qm'], 4, sys._getframe().f_code.co_name + ': full')
+
+
+def test_psi4_qm_1e():
```

Comment:
It would be be good to convert these to parametrize at some point.

Reply:
I've parametrized all the others, but `from_string` is on the messy side. saving it for a refactor/expansion.

Reply:
The new version looks really clean I think!